File: app/src/main/java/com/aaronbrecher/neverlate/adapters/EventDetailPagerAdapter.java
package com.aaronbrecher.neverlate.adapters;

import com.aaronbrecher.neverlate.ui.fragments.EventDetailFragment;
import com.aaronbrecher.neverlate.ui.fragments.EventDetailMapFragment;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

public class EventDetailPagerAdapter extends FragmentPagerAdapter {
    private int numTabs;

    public EventDetailPagerAdapter(FragmentManager fm, int numTabs) {
        super(fm);
        this.numTabs = numTabs;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new EventDetailFragment();
            case 1:
                return new EventDetailMapFragment();
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return numTabs;
    }
}

File: app/src/main/java/com/aaronbrecher/neverlate/adapters/EventDetailPagerAdapter.kt
package com.aaronbrecher.neverlate.adapters

import com.aaronbrecher.neverlate.ui.fragments.EventDetailFragment
import com.aaronbrecher.neverlate.ui.fragments.EventDetailMapFragment

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

class EventDetailPagerAdapter(fm: FragmentManager, private val numTabs: Int) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> EventDetailFragment()
            1 -> EventDetailMapFragment()
            else -> EventDetailFragment()
        }
    }

    override fun getCount(): Int {
        return numTabs
    }
}
